examples/c_api/deletes.c

```diff
@@ -0,0 +1,225 @@
+  tiledb_array_schema_t* array_schema;
+  tiledb_array_schema_alloc(ctx, TILEDB_SPARSE, &array_schema);
+  tiledb_array_schema_set_cell_order(ctx, array_schema, TILEDB_ROW_MAJOR);
+  tiledb_array_schema_set_tile_order(ctx, array_schema, TILEDB_ROW_MAJOR);
+  tiledb_array_schema_set_domain(ctx, array_schema, domain);
+  tiledb_array_schema_add_attribute(ctx, array_schema, a);
+
+  // Create array
+  tiledb_array_create(ctx, array_name, array_schema);
+
+  // Clean up
+  tiledb_attribute_free(&a);
+  tiledb_dimension_free(&d1);
+  tiledb_dimension_free(&d2);
+  tiledb_domain_free(&domain);
+  tiledb_array_schema_free(&array_schema);
+  tiledb_ctx_free(&ctx);
+}
+
+void write_array() {
+  // Create TileDB context
+  tiledb_ctx_t* ctx;
+  tiledb_ctx_alloc(NULL, &ctx);
+
+  // Open array for writing
+  tiledb_array_t* array;
+  tiledb_array_alloc(ctx, array_name, &array);
+  tiledb_array_open(ctx, array, TILEDB_WRITE);
+
+  // Write some simple data to cells (1, 1), (2, 4) and (2, 3).
+  int coords_rows[] = {1, 2, 2};
+  int coords_cols[] = {1, 4, 3};
+  uint64_t coords_size = sizeof(coords_rows);
+  int data[] = {1, 2, 3};
+  uint64_t data_size = sizeof(data);
+
+  // Create the query
+  tiledb_query_t* query;
+  tiledb_query_alloc(ctx, array, TILEDB_WRITE, &query);
+  tiledb_query_set_layout(ctx, query, TILEDB_UNORDERED);
+  tiledb_query_set_data_buffer(ctx, query, "a", data, &data_size);
+  tiledb_query_set_data_buffer(ctx, query, "rows", coords_rows, &coords_size);
+  tiledb_query_set_data_buffer(ctx, query, "cols", coords_cols, &coords_size);
+
+  // Submit query
+  tiledb_query_submit(ctx, query);
+
+  // Close array
+  tiledb_array_close(ctx, array);
+
+  // Clean up
+  tiledb_array_free(&array);
+  tiledb_query_free(&query);
+  tiledb_ctx_free(&ctx);
+}
+
+void delete_cells() {
+  // Create TileDB context
+  tiledb_ctx_t* ctx;
+  tiledb_ctx_alloc(NULL, &ctx);
+
+  // Delete cells where rows == 2.
+  tiledb_query_condition_t* qc;
+  tiledb_query_condition_alloc(ctx, &qc);
+  int val = 2;
+  tiledb_query_condition_init(ctx, qc, "rows", &val, sizeof(int), TILEDB_EQ);
+
+  // Open array for delete
+  tiledb_array_t* array;
+  tiledb_array_alloc(ctx, array_name, &array);
+  tiledb_array_open(ctx, array, TILEDB_DELETE);
+
+  // Create the query
+  tiledb_query_t* query;
+  tiledb_query_alloc(ctx, array, TILEDB_DELETE, &query);
+  tiledb_query_set_condition(ctx, query, qc);
+
+  // Submit query
+  tiledb_query_submit(ctx, query);
+
+  // Close array
+  tiledb_array_close(ctx, array);
+
+  // Clean up
+  tiledb_query_condition_free(&qc);
+  tiledb_array_free(&array);
+  tiledb_query_free(&query);
+  tiledb_ctx_free(&ctx);
+}
+
+void read_array() {
+  // Create TileDB context
+  tiledb_ctx_t* ctx;
+  tiledb_ctx_alloc(NULL, &ctx);
+
+  // Open array for reading
+  tiledb_array_t* array;
+  tiledb_array_alloc(ctx, array_name, &array);
+  tiledb_array_open(ctx, array, TILEDB_READ);
+
+  // Set maximum buffer sizes
+  uint64_t coords_size = 12;
```

Comment:
Can you be clearer to the reader on why is that `12` ? Maybe it's better to write `3*sizeof(int)` ?